### PR TITLE
docs: Remove api v2

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: docs: Remove api v2
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
@htuch as your comment in docs/build.sh suggests the biggest challenge to removing the v2 build is with the version history - i think that will require its own solution

there are a few more failures:

- api/xds_protocol.rst throws a ton of errors
- configuration/arch_overview/operations and maybe a few others - small number of errors

if we can agree a way forward for the version history (perhaps linking to old docs ?) then im happy to go through and update

not sure about the xds protocol doc - but i guess its a case of updating/rewriting it to point only to v3 xds

the other errors are pretty small in number, and im guessing we again just need to update the relevant docs

**Comment by htuch** [comment]
Yeah, for version history, let's replace these with HTTP references to the 1.17.0 v2 API docs. It should be possible to sed this, since the references follow a regular pattern, e.g. `envoy_api` vs. `envoy_v3_api`.

The rest of the docs should just point to the v3 counterparts.

**Comment by phlax** [comment]
docs here - https://storage.googleapis.com/envoy-pr/16077/docs/index.html

**Comment by htuch** [comment]
@phlax output looks good, happy to review when in ready state.

**Comment by phlax** [comment]
@htuch once #16155 has landed there will be one outstanding issue with this - the `not_so_simple_http_cache` issue  - once that is resolved we should be able to remove v2 from the docs

File: .azure-pipelines/pipelines.yml [hunk 1]
```diff
@@ -102,6 +102,7 @@ stages:
               BAZEL_REMOTE_INSTANCE: projects/envoy-ci/instances/default_instance
               GCP_SERVICE_ACCOUNT_KEY: $(GcpServiceAccountKey)
             displayName: "Generate docs"
+            continueOnError: true
```
Comment (phlax) [inline .azure-pipelines/pipelines.yml:105]: remove

**Comment by phlax** [comment]
@htuch this doesnt remove all of the v2 references - but it gets most of them and prevents the v2 proto -> rst build - so its a pretty good start i think

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Comment by alyssawilk** [comment]
hmmm, I noticed over on https://github.com/envoyproxy/envoy/pull/15926 if I fix_proto and fix_format it adds these back:

--- a/api/BUILD
+++ b/api/BUILD
@@ -133,12 +133,18 @@ proto_library(
         "//envoy/config/common/matcher/v3:pkg",
         "//envoy/config/core/v3:pkg",
         "//envoy/config/endpoint/v3:pkg",
+        "//envoy/config/filter/thrift/router/v2alpha1:pkg",
         "//envoy/config/grpc_credential/v3:pkg",
+        "//envoy/config/health_checker/redis/v2:pkg",
         "//envoy/config/listener/v3:pkg",
         "//envoy/config/metrics/v3:pkg",
         "//envoy/config/overload/v3:pkg",
         "//envoy/config/ratelimit/v3:pkg",
         "//envoy/config/rbac/v3:pkg",
+        "//envoy/config/resource_monitor/fixed_heap/v2alpha:pkg",
+        "//envoy/config/resource_monitor/injected_resource/v2alpha:pkg",
+        "//envoy/config/retry/omit_canary_hosts/v2:pkg",
+        "//envoy/config/retry/previous_hosts/v2:pkg",
         "//envoy/config/route/v3:pkg",
         "//envoy/config/tap/v3:pkg",
         "//envoy/config/trace/v3:pkg",

I'll just stash those changes and see if CI is happy, but I think we need to fix some auto-gen thing to avoid these being added back

**Comment by phlax** [comment]
@alyssawilk i think the fix is #16330

**Pull request merged**